lua/lualine/highlight.lua

```diff
@@ -10,7 +10,19 @@ local function highlight (name, foreground, background, special)
   return table.concat(command, ' ')
 end
 
+local function apply_defaults_to_theme(theme)
+  local modes = {'insert', 'visual', 'replace', 'command', 'terminal', 'inactive'}
+  for _, mode in ipairs(modes) do
+    theme[mode] = (theme[mode] or theme['normal'])
+    theme[mode]['a'] = (theme[mode]['a'] or theme['normal']['a'])
```

Comment:
This could be further cleaned up, especially when there is a possibility that there will be more section themes
Something like this could work
```lua
for section_name, section in pairs(theme['normal']) do
  theme[mode][section_name] = (theme[mode][section_name] or section)
end
```